### PR TITLE
Create UI Home Screen

### PR DESCRIPTION
**Pull request by tiennt-0361**
[Task][68400] Create UI
Create UI for Home screen
- List categories
- comming soon List favorite cooktail (use API to demo)
- Title label
- Search button
[Redmine link]
https://edu-redmine.sun-asterisk.vn/issues/68400
Screenshots (nếu có
Simulator Screen Shot - iPhone 11 - 2023-09-13 at 17 12 45
)
Screen Shot 2023-09-13 at 17 13 35

### CODE REVIEW

**Review by hlinh88 — APPROVED** [review]

File: CooktailDB/CooktailDB/ControllerView/ViewController.swift [hunk 1]
```diff
@@ -1,27 +1,89 @@
...
         fetchCooktail()
+        configCollectionView()
+    }
+    private func configCollectionView() {
+        if let layout = categoryCollectionView.collectionViewLayout as? UICollectionViewFlowLayout {
+            layout.scrollDirection = .horizontal
+        }
+        categoryCollectionView.register(
+            UINib(nibName: "CategoryViewCellCollectionViewCell", bundle: nil),
+            forCellWithReuseIdentifier: "CategoryCollectionViewCell")
```
Comment (ghost) [inline CooktailDB/CooktailDB/ControllerView/ViewController.swift]: constant vào

File: CooktailDB/CooktailDB/ControllerView/ViewController.swift [hunk 2]
```diff
@@ -1,27 +1,89 @@
...
+        if let layout = favoriteCollectionView.collectionViewLayout as? UICollectionViewFlowLayout {
+            layout.scrollDirection = .horizontal
+        }
+        favoriteCollectionView.register(UINib(nibName: "FavoriteCollectionViewCell", bundle: nil),
+                                        forCellWithReuseIdentifier: "FavoriteCollectionViewCell")
```
Comment (ghost) [inline CooktailDB/CooktailDB/ControllerView/ViewController.swift]: .

File: CooktailDB/CooktailDB/ControllerView/ViewController.swift [hunk 3]
```diff
@@ -1,27 +1,89 @@
...
+            return data.count
+        } else {
+            return cooktails.count
+        }
+    }
+    func collectionView(_ collectionView: UICollectionView,
+                        cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        if collectionView == currentCollectionView {
+            guard let cell = collectionView.dequeueReusableCell(
+                withReuseIdentifier: "CategoryCollectionViewCell",
```
Comment (ghost) [inline CooktailDB/CooktailDB/ControllerView/ViewController.swift]: .

File: CooktailDB/CooktailDB/ReuseableCell/CategoryViewCellCollectionViewCell.swift [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    }
+    private func config() {
+        layer.cornerRadius = Constant.CategoryCollectionViewConfig.backgrounConnerRadius
+        backgroundColor = UIColor(hex: "#FB7D8A")
+        categoryImage.layer.cornerRadius = Constant.CategoryCollectionViewConfig.imageConnerRadius
+    }
+    func setData(image: String, name: String, number: Int) {
+        self.categoryImage.image = UIImage(named: image)
+        self.categoryName.text = name
+        self.categoryNumber.text = "\(number + 1) mixes"
```
Comment (ghost) [inline CooktailDB/CooktailDB/ReuseableCell/CategoryViewCellCollectionViewCell.swift]:
```suggestion
    func setData(model: CategoryModel) {
        self.categoryImage.image = UIImage(named: model.image)
        self.categoryName.text = model.name
        self.categoryNumber.text = "\(model.number + 1) mixes"
```

File: CooktailDB/CooktailDB/ReuseableCell/FavoriteCollectionViewCell.swift [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        guard let cooktailImage = cooktail.cooktailImage else { return }
+        let safeUrl = cooktailImage + "/preview"
+        APIManager.shared.getImg(url: safeUrl) { [weak self] image in
+            guard let self = self else { return }
+            self.favoriteCooktailImage.image = image
+        }
+        favoriteCooktailName.text = cooktail.cooktailName
+        favoriteCooktailAlcoholic.text = cooktail.cooktailAlcoholic
+        cooktailCategory.text = cooktail.cooktailCategory
+    }
```
Comment (ghost) [inline CooktailDB/CooktailDB/ReuseableCell/FavoriteCollectionViewCell.swift:41]: đấy, làm như này này

**Review by ghost — APPROVED** [review]

**Pull request merged**